src/backend/env/mod.rs

```diff
@@ -271,6 +271,17 @@ pub enum Destination {
 }
 
 impl State {
+    pub fn register_post_tags(&mut self, post_id: PostId, tags: &BTreeSet<String>) {
+        for tag in tags {
```

Comment:
It's missing removing tags. If i remove tag from post, it still shows it

Reply:
That's fine, the deleted posts are not rendered in most feed anyway. also I think it's a good to see that the post existed. However, also note that tag indexes are rotating. They only keep the last 1k posts. So any de;eted post will disappear eventually anyway, this is why I see no reasons to complicate the code.

Reply:
But what about posts that are not deleted, but are edited to remove erroneously added tags? I think this is what @aligatorr89 was describing

Reply:
@taggrx , I described removing tags from post.